### PR TITLE
[PM-5796] Improve desktop biometric browser integration error handling

### PR DESCRIPTION
**Pull request by dani-garcia**
## Type of change
```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Enabling browser integration for biometric unlock in the desktop application requires the creation of certain manifest files in some OS and browser specific locations. These files are created only when the setting is enabled, and any errors that happen during this operation are simply logged to the console and the user is none the wiser.

This PR is an attempt to improve this situation in a couple of ways:
- We re-register these manifests at app startup, in case they were removed due to a browser reinstall or other causes.
- On Windows, we add the manifests unconditionally instead of trying to detect that the browser is installed. This caused some browser installs to be missed and the integration to fail (Windows Store/portable/system-wide installs).
- Adding some basic logging in the browser extension, where previously the error was ignored.
- Handling any errors that happen while enabling the setting on the desktop app, and surfacing a message to the user.

TODO: Oscar mentioned to me that we might want to offer a way to disable the startup manifest generation, to help when trying to test dev builds of the browser extensions.

## Code changes

- **file.ext:** Description of what was changed and why

## Screenshots
Added a single dialog:
image

### CODE REVIEW

File: apps/desktop/src/main.ts [hunk 1]
```diff
@@ -218,12 +218,17 @@ export class Main {
...
 
-        if (
-          (await this.stateService.getEnableBrowserIntegration()) ||
-          (await this.stateService.getEnableDuckDuckGoBrowserIntegration())
-        ) {
-          // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
-          // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        const browserIntegrationEnabled = await this.stateService.getEnableBrowserIntegration();
+        const ddgIntegrationEnabled =
+          await this.stateService.getEnableDuckDuckGoBrowserIntegration();
```
Comment (willmartian) [inline apps/desktop/src/main.ts:223]:
```suggestion
        const [browserIntegrationEnabled, ddgIntegrationEnabled] = await Promise.all([
          this.stateService.getEnableBrowserIntegration(),
          this.stateService.getEnableDuckDuckGoBrowserIntegration()
        ]);
```

File: apps/desktop/src/app/accounts/settings.component.ts [hunk 2]
```diff
@@ -643,11 +654,19 @@ export class SettingsComponent implements OnInit {
       await this.stateService.setDuckDuckGoSharedKey(null);
     }
 
-    this.messagingService.send(
-      this.form.value.enableDuckDuckGoBrowserIntegration
-        ? "enableDuckDuckGoBrowserIntegration"
-        : "disableDuckDuckGoBrowserIntegration",
+    const errorResult = await ipc.platform.nativeMessaging.manifests.generateDuckDuckGo(
+      this.form.value.enableDuckDuckGoBrowserIntegration,
     );
```
Comment (willmartian) [inline apps/desktop/src/app/accounts/settings.component.ts:659]: Is it good practice to explicitly reference something like `ipc.platform.nativeMessaging.manifests` directly in a component? This feels like something that would be abstracted behind an injected service.

**Review by willmartian — COMMENTED** [review]
I'm not familiar with this code domain, so I am going to pass to @justindbaur ✌️

File: apps/desktop/src/app/accounts/settings.component.ts (hunk 2 above)
Comment (dani-garcia) [reply apps/desktop/src/app/accounts/settings.component.ts]: That's a good point, I've extracted that code to a small service now.

**Review by justindbaur — DISMISSED** [review]

**Review by justindbaur — APPROVED** [review]

**Pull request merged**